blog/content/post/2021-12-12-optiplex-txt.md

```diff
@@ -0,0 +1,222 @@
+for your processor).
+
+You may also want to read [previous blog posts about Dell OptiPlex 7010 / 9010](https://blog.3mdeb.com/tags/optiplex/).
+
+## DRTM rationale
+
+Why do even Trusted Execution Technology and DRTM exist? First of all we have
+to start with the firmware, the inevitable piece of code initializing the
+silicon and other hardware. Its responsibility is to prepare the platform to
+run operating system, but also to load it. So basically the owner of the
+firmware has control what operating system can be launched, and what is most
+important, how it can be launched. By how we should understand whether the
+operating system is launched as intended without any undesired actions, like
+malware or spyware installation under the hood. In a case the attacker has
+malformed the firmware on your platform he may install key loggers and spying
+software to steal your data, passwords, etc. How to prevent that? How to ensure
+the operating system software was not tampered with? The answer is DRTM (
+Dynamic Root of Trust for Measurement). It allows to securely measure the
+operating system (calculate hash of its software components) in an unreplayable
+manner and store it inside a secure storage chip, i.e. the Trusted Platform
+Module.
+
+## Intel TXT deep dive
+
+Below diagram represents the typical boot flow of Intel TXT enabled machine
+with measured launch.
+
+![Intel TXT boot timeline](/img/txt_launch1.jpg)
+
+> Source: *[A Practical Guide to TPM 2.0](https://link.springer.com/book/10.1007/978-1-4302-6584-9)*
+
+First the BIOS is starting execution from the reset vector. It is responsible
+to initialize the Intel TXT to be used by operating system bootloader. For this
+purpose it calls the BIOS ACM (Authenticated Code Module) to properly set up
+Intel TXT. It must be noted that ACMs are signed by Intel and Intel CPU will
+only execute ACMs signed by Intel key. The key hash is burned into CPUs an is
+used to authenticate microcode updates and ACM signatures. Such measures ensure
+only the authorized code is able to initialize Intel TXT, which prevents its
+misconfiguration. When BIOS is done, it hands over the control to the operating
+system bootloader. Now it is the bootloader's responsibility to perform the
+measured launch of the operating system by calling the SINIT ACM:
+
+![Breakout of measured launch details](/img/txt_launch2.jpg)
+
+> Source: *[A Practical Guide to TPM 2.0](https://link.springer.com/book/10.1007/978-1-4302-6584-9)*
+
+Currently there are two methods of executing SINIT ACM to perform measured
+launch:
+
+- [Trusted Boot (tboot)](https://sourceforge.net/projects/tboot/) - Intel's
+  reference implementation of Measured Launch Environment for Intel TXT
+- [GRUB2 with TrenchBoot support](https://trenchboot.org/) - GRUB2 is the most
+  popular open source bootloader. It can integrate TrenchBoot to perform
+  measured launch with Intel TXT or AMD Secure Startup.
```

Comment:
There is very nice explanation what tboot really is and its persistence across plarform runtime in TB documentation.

Reply:
I can only see the explanation on recent slides: https://youtu.be/xZoCtNV8Qs0?t=12888
But I can't find anything like that on the TB documentation... Could you point me to the file(s)?

Reply:
This is documentation for linux docs: https://github.com/TrenchBoot/documentation/pull/16/files#diff-03b90ce541780e13410de22950e43562c58fe5f7625eceddb9b1ac8a06a96a0dR8

Reply:
Done